pkg/executor/analyze.go

```diff
@@ -393,70 +393,32 @@ func (e *AnalyzeExec) handleResultsError(
 	partitionStatsConcurrency = min(taskNum, partitionStatsConcurrency)
 	// If partitionStatsConcurrency > 1, we will try to demand extra session from Domain to save Analyze results in concurrency.
 	// If there is no extra session we can use, we will save analyze results in single-thread.
+	dom := domain.GetDomain(e.Ctx())
+	internalCtx := kv.WithInternalSourceType(ctx, kv.InternalTxnStats)
 	if partitionStatsConcurrency > 1 {
```

Comment:
```suggestion
	subSctxs := dom.FetchAnalyzeExec(partitionStatsConcurrency) // No matter 1 or more than 1
	logutil.BgLogger().Info("use multiple sessions to save analyze results", zap.Int("sessionCount", len(subSctxs)))
	if len(subSctxs) > 0 {		
			defer func() {
				dom.ReleaseAnalyzeExec(subSctxs)
			}()
      }
      return e.handleResultsErrorWithConcurrency(internalCtx, concurrency, needGlobalStats, subSctxs, globalStatsMap, resultsCh)
	
```

Reply:
We cannot do this because this is no guarantee that we can get any sessions from `FetchAnalyzeExec`.